src/ty_check/mod.rs

```diff
@@ -821,9 +823,17 @@ impl<'a> TyChecker<'a> {
                     )));
                 }
             }
+            TyKind::Data(DataTy::Ref(_, _, _, arr_ty)) => match *arr_ty {
+                DataTy::Array(elem_ty, n) => (*elem_ty, n),
+                _ => {
+                    return Err(self.ty_error(TyErrorKind::String(
+                        "Trying to index into non array type1.".to_string(),
```

Comment:
Do we need the `1` in `type1` in both error messages?

Reply:
I added this to distinguish different panics with the same error message I will delete this